src/controllers/root.controller.js

```diff
@@ -17,26 +11,7 @@ module.exports = class RootController extends BaseController {
   }
 
   static async doPost (request, reply, errors) {
-    // Generate a session token
-    const token = uuid4()
-
-    // Generate a CRM token
-
-    let authToken
-    try {
-      authToken = await authService.getToken()
-    } catch (err) {
-      console.error(err)
-      return reply.redirect('/error')
-    }
-
-    // TODO: Confirm how session handling will work and where the most
-    // appropriate point is to create and destroy session cookies
-
-    const cookie = {
-      token: token,
-      authToken: authToken
-    }
+    const cookie = await this.generateCookie(reply)
```

Comment:
Maybe another one for discussion but I think I would rather specify the class name when calling static methods, so use **BaseController**.generateCookie rather than **this**.generateCookie

I think maybe it is clearer where the static method is that is being called.

If we keep this as-is then we probably want to change all the BaseController.createPageContext calls to keep it consistent.

Reply:
I don't have a strong opinion on this so happy to change it – @Cruikshanks, do you have a view?

Reply:
Might be my back ground, but would go with using the class name rather than `this`. For me when I see `this` my automatic assumption is you are in the context of an instance.

Reply:
Cool, I think we're all agreed then! Thanks @duncannixon and @Cruikshanks, I'll get that updated.